pkg/cloud/gcpprovider.go

```diff
@@ -1349,12 +1356,27 @@ func (gcp *GCP) AllNodePricing() (interface{}, error) {
 // NodePricing returns GCP pricing data for a single node
 func (gcp *GCP) NodePricing(key Key) (*Node, error) {
 	gcp.DownloadPricingDataLock.RLock()
-	defer gcp.DownloadPricingDataLock.RUnlock()
 	if n, ok := gcp.Pricing[key.Features()]; ok {
 		klog.V(4).Infof("Returning pricing for node %s: %+v from SKU %s", key, n.Node, n.Name)
 		n.Node.BaseCPUPrice = gcp.BaseCPUPrice
+		gcp.DownloadPricingDataLock.RUnlock()
 		return n.Node, nil
+	} else if _, ok := gcp.ValidPricingKeys[key.Features()]; ok {
+		gcp.DownloadPricingDataLock.RUnlock()
+		err := gcp.DownloadPricingData()
```

Comment:
This is the biggest problem with go RWMutexes is there's not API support to upgrade the lock from read to readwrite, so this "gap" where we unlock the read and then lock read/write leaves room for something else to potentially take the lock before executing `DownloadPricingData()`. 

Given the circumstances and likely low contention on that specific lock, this bit of code is likely ok, as I don't think that anything stealing the lock between these calls will create any unhealthy scenarios in `DownloadPricingData()`.